### ISSUE TITLE
Allow to extend on existing Pipelines

### ISSUE THREAD

**Issue by EagleoutIce** [Code-Inspect/flowr#656]
Currently the `PipelineExecutor` is a one-shot kind of system.
Yet, what if we, for example want to parse a file and then run both the old and the new dataflow, comparing their result. 
A nice `PipelineExecutor` could cache the results of the parse step as it is the same for both.

**Comment by bjthehun** [comment]
#280 also talks about caching of results. How do these issues differ?

**Comment by EagleoutIce** [comment]
Good catch! This is essentially a "newer" version which wants much more flexible caching with the help of the PipelineExecutor of #526.

**Comment by MaxAtoms** [comment]
Do we fulfill this with the `flowr-analyzer-cache.ts` in combination with #1942?
